dev-docs/howtoguides/how_to_setup_windows_machine_for_wsl_testing.md

```diff
@@ -0,0 +1,158 @@
+# How to setup Windows machine for WSL testing
+
+To run our integration tests on WSL instances, we need to first setup a Windows machine
+to allows us to launch the WSL instances. Not only that, we also need to setup OpenSSH
+server to allow us to run ssh commands on the Windows machine.
+
+This how to guide will explain all the necessary steps for the setup, from launching the image
+from Azure to testing if WSL is correctly working on the machine.
+
+## Launching the image on Azure
+
+When launching the Windows instance on Azure, there a few configurations that need to be performed,
+otherwise we will not be able to launch WSL instances.
+
+First of all, name the machine as `wsl-test`. Currently, this is the name we expect on the
+integration tests.
+
+After that, select the `Windows 11` image and mark the `Security Type` as *Standard*.
+![windows_secuirty_type](windows-security-type.png)
+
+Set an username and password for the VM. Please set the username as *ubuntu*. 
+We will use this account setting to manually login into the machine to run some commands there directly.
+![windows_secuirty_type](windows-ssh-setup.png)
+
+Finally, remember to also select the SSH inbound port for the machine
+![windows_secuirty_type](windows-setup-port.png)
+
+
+## Configure Bastion to login into the machine
+
+To login into the machine and access its GUI, you need to setup Bastion.
+After launching the machine, go for the *Connect with Bastion* option:
+![windows_secuirty_type](windows-bastion.png)
+![windows_secuirty_type](windows-deploy-bastion.png)
+
+After deploying Bastion, you just need to present your username and password to login into the
+machine.
+
+## Configuring Hyper-V
+
+After login into the machine through Bastion, you need to configure Hyper-V by following
+this [tutorial](https://learn.microsoft.com/en-us/azure/lab-services/how-to-enable-nested-virtualization-template-vm-using-script?tabs=powershell)
```

Comment:
I'm missing a reference to the `Virtual Machine Platform` optional feature. Does this step cover it?

We had this error when setting up a VM, fixed by setting that feature ON, if you remember:

```
WslRegisterDistribution failed with error: 0x80070424
Error: 0x80070424 The specified service does not exist as an installed service.
```

Reply:
What I have found is that when just running `wsl --install`, it already covers this step automatically.
But I will let @orndorffgrant and @renanrodrigo try this out first just to guarantee if this is indeed correct

Reply:
According to the Microsoft Docs, `--install` does it for you. https://learn.microsoft.com/en-us/windows/wsl/install#install-wsl-command.

@CarlosNihelton and I are just not used to these new conveniences 😅.

Reply:
Indeed 😄